backends/aerospike.go

```diff
@@ -65,9 +65,13 @@ func NewAerospikeBackend(cfg config.Aerospike, metrics *metrics.Metrics) *Aerosp
 
 	client, err := as.NewClientWithPolicyAndHost(clientPolicy, hosts...)
 	if err != nil {
-		log.Fatalf("%v", formatAerospikeError(err).Error())
+		log.Fatalf("%v", classifyAerospikeError(err).Error())
 		panic("AerospikeBackend failure. This shouldn't happen.")
 	}
+
+	// If the record already exists, don't write and throw error
+	client.DefaultWritePolicy.RecordExistsAction = as.CREATE_ONLY
```

Comment:
Should this be defined in `clientPolicy` a few lines up?

Reply:
It is better to do it here because `NewClientWithPolicyAndHost` returns a client with a generic `DefaultWritePolicy` in line 98 below:
```
 80 func NewClientWithPolicyAndHost(policy *ClientPolicy, hosts ...*Host) (*Client, error) {
 81     if policy == nil {
 82         policy = NewClientPolicy()
 83     }
 84 +-- 10 lines: cluster, err := NewCluster(policy, hosts)----------------------------------------------------
 94     client := &Client{
 95         cluster:            cluster,
 96         DefaultPolicy:      NewPolicy(),
 97         DefaultBatchPolicy: NewBatchPolicy(),
 98         DefaultWritePolicy: NewWritePolicy(0, 0),
 99         DefaultScanPolicy:  NewScanPolicy(),
100         DefaultQueryPolicy: NewQueryPolicy(),
101         DefaultAdminPolicy: NewAdminPolicy(),
102     }
103
104     runtime.SetFinalizer(client, clientFinalizer)
105     return client, err
106
107 }
108
109 //-------------------------------------------------------
110 // Cluster Connection Management
~/go/pkg/mod/github.com/aerospike/aerospike-client-go@v4.0.0+incompatible/client.go [RO]
```